java/steps/yaks-jms/src/main/java/org/citrusframework/yaks/jms/JmsSteps.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.citrusframework.yaks.jms;
+
+import static com.consol.citrus.actions.ReceiveMessageAction.Builder.receive;
+import static com.consol.citrus.actions.SendMessageAction.Builder.send;
+
+import org.springframework.util.Assert;
+
+import com.consol.citrus.Citrus;
+import com.consol.citrus.TestCaseRunner;
+import com.consol.citrus.annotations.CitrusFramework;
+import com.consol.citrus.annotations.CitrusResource;
+import com.consol.citrus.dsl.endpoint.CitrusEndpoints;
+import com.consol.citrus.jms.endpoint.JmsEndpoint;
+import com.consol.citrus.jms.message.JmsMessage;
+
+import javax.jms.ConnectionFactory;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import io.cucumber.datatable.DataTable;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+public class JmsSteps {
+
+    private static final long TIMEOUT = System.getenv("YAKS_JMS_TIMEOUT") != null ? Integer.valueOf(System.getenv("YAKS_JMS_TIMEOUT")) : TimeUnit.SECONDS.toMillis(60);
+
+    @CitrusResource
+    private TestCaseRunner runner;
+
+    @CitrusFramework
+    private Citrus citrus;
+
+    private JmsEndpoint jmsEndpoint;
+
+    private ConnectionFactory connectionFactory;
+
+    @Given("^(?:C|c)onnection factory class: (.+)$")
```

Comment:
`class` may be too technical in terms of BDD and can be removed in my eyes

Reply:
Also I think we should include `JMS` because there might be other connection factories that we want to initialize in other modules and this would then collide with the same wording in two different modules.

So we end up with

```
Given JMS connection factory
```